internal/hcs/process.go

```diff
@@ -167,7 +167,38 @@ func (process *Process) Kill(ctx context.Context) (bool, error) {
 		return true, nil
 	}
 
-	resultJSON, err := vmcompute.HcsTerminateProcess(ctx, process.handle)
+	// HCS serializes the signals sent to a target pid per compute system handle.
+	// To avoid SIGKILL being serialized behind other signals, we open a new compute
+	// system handle to deliver the kill signal.
+	// If the calls to opening a new compute system handle fail, we forcefully
+	// terminate the container itself so that no container is left behind
+	hcsSystem, err := OpenComputeSystem(ctx, process.system.id)
```

Comment:
If possible, it would be great if you can do a manual test of the error path here by injecting an error and make sure the compute system is properly cleaned up.

Reply:
Will try this now

Reply:
I was able to validate this. I found out that line 186 was mistakenly removed in the previous commit during dead code cleanup. Brought that in and validated the fix